c/include/proton/annotations.h

```diff
@@ -90,4 +90,24 @@
     #endif
 #endif
 
+// fallthrough
+#if defined __cplusplus && defined __has_cpp_attribute
+    #if __has_cpp_attribute(fallthrough) && __cplusplus >= __has_cpp_attribute(fallthrough)
+        #define PN_FALLTHROUGH [[fallthrough]]
+    #endif
+#endif
+#if !defined PN_FALLTHROUGH && defined __STDC_VERSION__ && defined __has_c_attribute
+    #if __has_c_attribute(fallthrough) && __STDC_VERSION__ >= __has_c_attribute(fallthrough)
+        #define PN_FALLTHROUGH [[fallthrough]]
+    #endif
+#endif
+#if !defined PN_FALLTHROUGH && defined __has_attribute
```

Comment:
again could be `#elif`

Reply:
no, because we might be on older c that has a __has_c_attribute but not the [[fallthrough]];, so we would go into

```
#if !defined PN_FALLTHROUGH && defined __STDC_VERSION__ && defined __has_c_attribute
````

but not define PN_FALLTHROUGH there